universal-login-wallet/src/ui/react/Modals/Transfer/ModalTransfer.tsx

```diff
@@ -16,9 +16,11 @@ const ModalTransfer = () => {
 
   const transferService = new TransferService(sdk, applicationWallet);
   const onGenerateClick = async () => {
-    modalService.showModal('waitingForTransfer');
     try {
-      await transferService.transfer(transferDetalis);
+      const {waitToBeSuccess, waitForTransactionHash} = await transferService.transfer(transferDetalis);
+      const {transactionHash} = await waitForTransactionHash();
+      modalService.showModal('waitingForTransfer', transactionHash);
```

Comment:
We should show dialog before we have a hash (but without a hash), and update once we have a hash